src/Raven.Client/Documents/DocumentStore.cs

```diff
@@ -344,24 +345,74 @@ public override IDisposable AggressivelyCacheFor(TimeSpan cacheDuration, string
         /// without touching the server.
         /// </remarks>
         public override IDisposable AggressivelyCacheFor(TimeSpan cacheDuration, AggressiveCacheMode mode, string database = null)
+        {
+            var release = SetAggressiveCache(cacheDuration, mode, ref database);
+
+            return AsyncHelpers.RunSync(() => FinalizeAggressiveCacheAsync(release, mode, database));
```

Comment:
Is there a reason why the helper doesn't provide a state-based overload that would avoid the closure?